packages/pluggableWidgets/datagrid-dropdown-filter-web/src/components/__tests__/FilterComponent.spec.tsx

```diff
@@ -0,0 +1,212 @@
+                        filterDispatcher={jest.fn()}
+                    />
+                );
+
+                expect(component).toMatchSnapshot();
+                expect(component.find("input").first().prop("value")).toBe("find me");
+            });
+        });
+
+        describe("with default options set", () => {
+            it("selects single default option", () => {
+                const defaultOption = defaultOptions[0];
+                const component = shallow(
+                    <FilterComponent
+                        multiSelect
+                        options={defaultOptions}
+                        filterDispatcher={jest.fn()}
+                        defaultValue={defaultOption.value}
+                    />
+                );
+
+                const input = component.find("input").first();
+
+                expect(input.prop("value")).toBe(defaultOption.caption);
+            });
+
+            it("selects multiple default options", () => {
+                const defaultValue = `${defaultOptions[0].value},${defaultOptions[1].value}`;
+
+                const component = shallow(
+                    <FilterComponent
+                        multiSelect
+                        options={defaultOptions}
+                        filterDispatcher={jest.fn()}
+                        defaultValue={defaultValue}
+                    />
+                );
+
+                const input = component.find("input").first();
+                const expectedCaptions = `${defaultOptions[0].caption},${defaultOptions[1].caption}`;
+                expect(input.prop("value")).toBe(expectedCaptions);
+            });
+
+            it("filters incorrect default options", () => {
+                const inCorrectDefaultValue = `${defaultOptions[0].value},${defaultOptions[1].value},SomeRandomText`;
+
+                const component = shallow(
+                    <FilterComponent
+                        multiSelect
+                        options={defaultOptions}
+                        filterDispatcher={jest.fn()}
+                        defaultValue={inCorrectDefaultValue}
+                    />
+                );
+
+                const input = component.find("input").first();
+
+                const expectedCaptions = `${defaultOptions[0].caption},${defaultOptions[1].caption}`;
+                expect(input.prop("value")).toBe(expectedCaptions);
+            });
+        });
+
+        describe("when value changes", () => {
+            it("calls filterDispatcher when value changes", () => {
+                const filterDispatcher = jest.fn();
+                const component = shallow(
+                    <FilterComponent multiSelect options={defaultOptions} filterDispatcher={filterDispatcher} />
+                );
+
+                const item = component.find("li").first();
+                item.simulate("click");
+
+                expect(filterDispatcher).toBeCalled();
```

Comment:
Ditto as above regarding testing args; is there value in it?

Reply:
Already answered. e2e tests it is